proposal.rst

```diff
@@ -260,18 +351,53 @@ Project Timeline
 .. and/or exams in this time, so we certainly aren't expecting full time work
 .. on your project, but you should aim to complete preliminary work such that
 .. you can actually start coding at the start of the coding period.
-.. 
+..
 .. The coding period is broken into three blocks of about 4 weeks each, with
 .. an evaluation after each block.  The evaluations are to help keep you on
 .. track, and consist of brief evaluation forms sent to GSoC by both the
 .. student and the mentor, and a chance to explicitly review how your project
 .. is going with Xapian mentors.
-.. 
+..
 .. If you will have other commitments during the project time (for example,
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+
+**Community Bonding Period: (May 4 - June 1) Module 1,2 **
+
+  * Get to know the community and interact with its members
+  * Understand Xapian Code Base, coding conventions and all relevant classes
+  * Understand how the code I am going to add will be structured and what kind of tests i will have to write
+  * Understand which file formats were chosen for previous year's project and why and understand how they were integrated
+  * Discuss with community, other than zip, which file formats are important and are needed to be added to omega.
+
+
+**Coding Weeks 1-3(June 1 - June 20) Module 3**
+
+  * Research about different zip file reading libraries- eg libarchive,libzip ( Compare them and discuss with the community which is the most appropriate)
```

Comment:
I'd encourage you to try to get the bulk of such research done before the coding period starts.

And things involving community discussion will tend to take a lot of elapsed time, so better to start them sooner.

Reply:
I guess I underestimated the time it would take to discuss which zip file reading libraries are useful for the community and I’m grateful for your insight that it could take longer. I’m now making this the twin primary focus of the Community Bonding Period along with a project-specific first PR. And I’ve updated the timeline to reflect some of this.

Reply:
It was really a general point about getting these sort of decisions done early, and about community discussion - in the past I've seen students surprised by how long it takes in elapsed time to consult the wider community on things.

In this particular case I'd probably suggest to go with libarchive, since it's mature, widely used, I've used it myself in the past with success, and it supports other archive formats beyond zip files (and being able to index files inside various archive formats would be useful for some people).